### PR TITLE
[#37] Add LibChat

### LINKED ISSUE #37: Add Libchat
Same as on the galter website. 

See https://github.com/galterlibrary/galter-website/issues/2731

### PR DESCRIPTION
**Pull request by sharpattack**
Adds LibChat with floating button, same as it is on the galter website. closes #37

### CODE REVIEW

**Comment by sharpattack** [comment]
Please message me if you would like to see how it looks with the chat available.

**Review by carsonicator — APPROVED** [review]
Thanks @sharpattack I connected with Corinne via the chat widget and it seems to working fine.

File: primo-explore/custom/main.js [hunk 1]
```diff
@@ -639,15 +633,23 @@ app.component('prmFullViewAfter', {
...
+});
+
+app.component('prmExploreFooterAfter', {
+  bindings: { parentCtrl: `<` },
+  template: `<lib-chat-component></lib-chat-component>`
+});
+
+(function () {
+  var lc = document.createElement('script');lc.type = 'text/javascript';lc.async = 'true';
+  lc.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'galter-northwestern.libanswers.com/load_chat.php?hash=4e6bc50e0cea1aa526c386eb796636d7';
```
Comment (Meowcenary) [inline primo-explore/custom/main.js:650]: What is "4e6bc50e0cea1aa526c386eb796636d7" ? I'm guessing it's an API key/token or something similar. Personally I think this kind of thing is nice to throw in a variable with a descriptive name, but maybe it's clear what's going on if you look into the LibChat documentation.

**Review by Meowcenary — APPROVED** [review]
One small comment, but nothing that should stop this from going in.

File: primo-explore/custom/main.js (hunk 1 above)
Comment (mattbaumann1) [reply primo-explore/custom/main.js:650]: agreed

**Review by mattbaumann1 — APPROVED** [review]
I would agree with @Meowcenary if that is a token that it belongs in a variable and probably the commits etc. should be adjusted so that it never shows up in the repo (if that is what's going on there).

Besides that, this looks good.

**Comment by sharpattack** [comment]
Sorry it took some time, but I moved the LibChat ID to an env variable, set in a `config.js` file that is now excluded from git.

File: primo-explore/custom/main.js [hunk 2]
```diff
@@ -639,15 +634,28 @@ app.component('prmFullViewAfter', {
...
-//   lc.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'northwestern.libanswers.com/load_chat.php?hash=8479f851eb63bff929998c6128b41b20';
-//   var s = document.getElementsByTagName('script')[0];s.parentNode.insertBefore(lc, s);
-//
-//   // End the chat button
-// })();
+// LibChat button
+app.component('libChatComponent', {
+  controller: [function() {
+    this.$onInit = function(){
+      this.libchatId = config.LIBCHAT_ID;
```
Comment (Meowcenary) [inline primo-explore/custom/main.js:641]: nice 😎👍

**Review by Meowcenary — APPROVED** [review]
Re-approving. Thanks for the change!

How about those Phillies tonight 👀

**Pull request merged**